torch/_inductor/shape_propagation.py

```diff
@@ -10,6 +10,16 @@
 
 
 BlockShapeType = Optional[Sequence[Union[int, str]]]
+SymBlockShapeType = Optional[Sequence[Union[int, str, sympy.Expr]]]
+
+
+def cast_sym_block_shape(sym_block_shape: SymBlockShapeType) -> BlockShapeType:
```

Comment:
cc @isuruf, can we add a codegen checking of shapes and enable it in opinfos ? see: https://github.com/pytorch/pytorch/blob/410ed3006b096753f388b42fd532d7b0e94524fa/test/inductor/test_torchinductor_opinfo.py#L1150

Reply:
Yes, I'll add a `runtime_triton_shape_assert`.